File: 3.x.x/crates-utopia/span/src/exporter/none.rs
use crate::exporter::Exporter;
use crate::span::Span;

/// Null exporter that discards all spans (PHP `Exporter\None`).
#[derive(Debug, Default, Clone, Copy)]
pub struct None;

impl None {
    pub fn new() -> Self {
        Self
    }
}

impl Exporter for None {
    fn sample(&self, _span: &Span) -> bool {
        false
    }

    fn export(&self, _span: &Span) {}
}

File: packages/span/src/Span/Exporter/None.php
<?php

declare(strict_types=1);

namespace Utopia\Span\Exporter;

use Utopia\Span\Span;

/**
 * Null exporter that discards all spans
 * Useful for testing or disabling tracing
 */
class None implements Exporter
{
    public function sample(Span $span): bool
    {
        return false;
    }

    public function export(Span $span): void
    {
        // Intentionally empty - discards all spans
    }
}
